### PR TITLE
Add blogpost about Numba dynamic exceptions

### PR DESCRIPTION
**Pull request by guilhermeleobas**
## Text styling

- [x] The blog is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [x] If there are headers, they use the proper header tags in order to do so (with only one level-one header).
- [x] All links describe where they link to (for example, check the [Quansight labs website](https://labs.quansight.org/)).
- [x] Any kind of styling that the author uses (for example, bold for emphasis) is consistent throughout the blog.

## Non-text contents

- [x] All content is represented as text (for example, images need alt text and videos need captions or descriptive transcripts).
- [x] If there are emojis, there are not more than three in a row.
- [x] Don't use [flashing gifs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [x] If it were to be read as plain text, the blog still makes sense and no information is missing.

### CODE REVIEW

File: apps/labs/posts/numba-dynamic-exceptions.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+To support dynamic exceptions, we reuse all the existing fields and introduce two new ones.
+
+- A pointer (`i8*`) to a pickled string containing static information
+- String size (`i32`)
+- The third argument (`i8*`), which was previously used for hashing is now used to hold a list of native values
+- A pointer to a function (`i8*`) that knows how to convert native values back to Python values. This is called [boxing](https://numba.pydata.org/numba-doc/dev/extending/interval-example.html#boxing-and-unboxing).
+- A flag (`i32`) to signal whether an exception is static or dynamic. A value greater than zero not only indicates whether it is a dynamic exception, but also the number of runtime arguments
+
+Using Python code, dynamic exceptions work as follow:
```
Comment (rgommers) [inline apps/labs/posts/numba-dynamic-exceptions.md:132]: typo: "as follows"

File: apps/labs/posts/numba-dynamic-exceptions.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+### Dynamic Exceptions
+
+To support dynamic exceptions, we reuse all the existing fields and introduce two new ones.
+
+- A pointer (`i8*`) to a pickled string containing static information
+- String size (`i32`)
+- The third argument (`i8*`), which was previously used for hashing is now used to hold a list of native values
+- A pointer to a function (`i8*`) that knows how to convert native values back to Python values. This is called [boxing](https://numba.pydata.org/numba-doc/dev/extending/interval-example.html#boxing-and-unboxing).
+- A flag (`i32`) to signal whether an exception is static or dynamic. A value greater than zero not only indicates whether it is a dynamic exception, but also the number of runtime arguments
```
Comment (rgommers) [inline apps/labs/posts/numba-dynamic-exceptions.md:130]: typo: missing period at end of sentence

File: apps/labs/posts/numba-dynamic-exceptions.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+During runtime, just before the control flow is returned to the interpreter, function `__exc_conv` is invoked to convert native `string/int` values into their equivalent Python `str/int` types. At this stage, the interpreter also unpickles constant information, and both static and dynamic arguments are combined into a unified list (3).
+
+I encourage anyone interested in further details to read the comments left on `callconv.py::CPUCallConv` ([ref](https://github.com/numba/numba/blob/c9cc06ba1410aff242764ffde8387a1bef2180ae/numba/core/callconv.py#L411-L444)).
+
+
+(3) https://github.com/numba/numba/blob/82d3cbb8818b43dc66e5dd4bb38355eaf25131be/numba/core/serialize.py#L64-L73
+
+## Limitations and future work
+
+Numba has a [page](https://numba.readthedocs.io/en/stable/reference/pysupported.html#exception-handling) describing what is supported in exception handling. Some work still needs to be done to support exceptions to its full extent.
```
Comment (rgommers) [inline apps/labs/posts/numba-dynamic-exceptions.md:169]: minor textual: end with "to their full extent."

File: apps/labs/posts/numba-dynamic-exceptions.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+We would like to thank Bodo for sponsoring this work and the Numba core developers and community for reviewing this work and the useful insights given during code review.
```
Comment (rgommers) [inline apps/labs/posts/numba-dynamic-exceptions.md:171]: Can you make `Bodo` a link (https://bodo.ai)?

**Review by rgommers — APPROVED** [review]
This is a really well-written blog post, thanks @guilhermeleobas! The structure is great, I only have very minor suggestions.

File: apps/labs/posts/numba-dynamic-exceptions.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+- A list of dynamic arguments: `[native string, int64]`
+- A pointer to `__exc_conv`
+- Number of dynamic arguments: `2`
+
+During runtime, just before the control flow is returned to the interpreter, function `__exc_conv` is invoked to convert native `string/int` values into their equivalent Python `str/int` types. At this stage, the interpreter also unpickles constant information, and both static and dynamic arguments are combined into a unified list (3).
+
+I encourage anyone interested in further details to read the comments left on `callconv.py::CPUCallConv` ([ref](https://github.com/numba/numba/blob/c9cc06ba1410aff242764ffde8387a1bef2180ae/numba/core/callconv.py#L411-L444)).
+
+
+(3) https://github.com/numba/numba/blob/82d3cbb8818b43dc66e5dd4bb38355eaf25131be/numba/core/serialize.py#L64-L73
```
Comment (kgryte) [inline apps/labs/posts/numba-dynamic-exceptions.md:165]: Should this link be here? It is included in the references section below.

**Review by kgryte — APPROVED** [review]

**Comment by rgommers** [comment]
@pavithraes @trallard @bskinn not sure what the way to get this published is now, can one of you comment/help? It should be ready to go.

**Comment by trallard** [comment]
If all looks good on the preview we can merge it now. 
If there is no rush for publishing we can wait until the usual release on Tuesday or do an ad-hoc one.

**Pull request merged**